types/node/events.d.ts

```diff
@@ -585,16 +635,42 @@ declare module "events" {
              */
             readonly asyncResource: EventEmitterReferencingAsyncResource;
         }
+
+        export interface EventEmitterBuiltInEventMap {
+            newListener: [eventName: string | symbol, listener: Function];
+            removeListener: [eventName: string | symbol, listener: Function];
+        }
     }
     global {
         namespace NodeJS {
-            interface EventEmitter<T extends EventMap<T> = DefaultEventMap> {
-                [EventEmitter.captureRejectionSymbol]?<K>(error: Error, event: Key<K, T>, ...args: Args<K, T>): void;
+            interface EventEmitter<Events extends EventMap<Events> = {}> {
+                // This "property" is used to brand a specific instance of the EventEmitter class with its Event map, which is needed
+                // in order to infer the map if we have a chain like `class A extends EventEmitter<{}>` (or many levels deep)
+                // It is also marked as possibly undefined in order to allow something like `const t: NodeJS.EventEmitter<{}> = { <insert implementation here> };`
+                readonly [internalTypeOnlyBrandSymbol]?: Events;
```

Comment:
This does hack around the inference problem, but it'd be worth getting an opinion from a DT maintainer on whether adding non-existent pseudoproperties is OK in a critical package. Even though it's defined as optional, it's not completely free of side-effects (pollutes `keyof EventEmitter`, for example).

Reply:
presumably this needs to be thoroughly looked at by DT maintainers anyways.
Anyways, the options are
1. Keep it like this
2. Revert this (leave the tests and adjust expect errors as necessary)
2a. Have implementers encourage passing the generics manually
2b. Allow typescript to infer this in a way that makes sense
3. Not merging this due to this impossibility.

Worth noting, it's always been easy to override the types for the instance methods when extending event emitter, but is recently completely impossible to get strictly typed static methods. If we don't have generics / strict type options on the static methods, that suggests to me a re-evaluation of whether they should exist on the base instance methods too.

Reply:
The best case scenario here is to only keep test changes for future since you have invested in them. The change is too breaky and incomplete to take in at this point

Reply:
What's incomplete about it? It was incomplete before. 
And to be clear, are you suggesting keeping strictly only the test changes? Because a lot of those won't work without my other changes.

Reply:
Incomplete as in falsifying data by adding properties that dont exist. 

This is too much of a breaking change to take it in and we would not be taking it without evidence of it helping a lot of users. This will be merged only if you want to add some tests for future without actual changes.

Reply:
I'm confused by the breaking part. <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68313> is the one that would be more "breaking" in my mind. This PR actually aligns more with the actual implementation while keeping the benefits that PR brought. I can certainly remove the symbol and make it so that where it does improperly infer it will fallback to the original behavior.